### ISSUE TITLE
Update the SingleR workflow to take as input a single model and output a single set of annotations using label.ont

### ISSUE THREAD

**Issue by allyhawkins** [AlexsLemonade/scpca-nf#365]
We have decided to update our usage of `SingleR` so that we annotate each SCE object with a single reference and single set of labels. The output would be a single column with cell type annotations. Currently, the workflow is designed to take as input a list of reference models to use for annotation. We should make the necessary changes so the workflow expects only a single model. 

One thing to note is that the reference files we use as input to the classification process, contain a list of objects. Each object contains the trained reference for `label.main`, `label.ont`, and `label.fine`. Right now we read in that file and perform annotation for all models and all label types. We can keep this organization, but then we will return three columns with cell type annotations (one for each label). I think we should simplify this and only use `label.ont` for cell type annotation. We can then get the cell type labels associated with that label using `ontoProc`. 

The reference files are created by a separate workflow that builds the references, outputting a single file containing three objects - one for each label type. I think we could keep the process we have to build references as is and then just grab the `label.ont` after reading the file into R. Or we could also update the process for building the cell type references so that each file only contains a single model built from a single label type.

**Comment by sjspielman** [comment]
I've spent some time familiarizing myself with the workflow, which was good to do in-and-of-itself! I'm thinking...
- Let's just grab `label.ont` from R. We can also add an argument to the R script indicating which label to return with a default of `"label.ont"`
- In terms of providing a single model, there are 2 main ways I can see to do this:
  - We could directly updated the `scpca-project-celltype-metadata.tsv` file itself, but with just one line per `scpca_project_id`.  This seems to be the right move imo? Or at least better than below..
  - We could keep `scpca-project-celltype-metadata.tsv` as-is, but _also_ somehow provide information to the workflow about which model in that file we'd want to use. That would probably mean modifying the library metadata file, which I'm rather hesitant to do... 

@allyhawkins, any feedback here? Is there another approach here I'm not immediately seeing that you are thinking of? Thanks!

**Comment by allyhawkins** [comment]
> Let's just grab `label.ont` from R. We can also add an argument to the R script indicating which label to return with a default of `"label.ont"`
> * (Edit) - The R script could also be updated such that it only _checks_ for the given label to return. Doesn't need to check for labels that won't be used.

Yes, the main change I would like to see is that we only output `label.ont` here. So that means the models that get read in should either only have `label.ont,` or we have to read them in and only grab the `label.ont` model. 
For some additional context, when we create the models they are stored as a list of models within a single `rds` file. This means when we read in the reference model for `HumanPrimaryCellAtlasData` we are reading in three models. This is why there is a lot of `purrr` happening in the `classify_SingleR.R` script. I think we could do one of the following: 

- update the original reference building to contain a `label_type` option and only train a model using one label, not all three. By default, this should be `label.ont`. Then the only changes that need to be made in `classify_SingleR.R` will be to accommodate reading in `rds` files that contain a single model object rather than 3 of them. 
- Keep the references the same and then specifically only grab the `label.ont` model and use that for classifying cells. 

I think I favor changing it at the stage where we build the reference models. I think this would simplify everything so that we only work with the `label.ont` moving forward, which is what we want. 

An additional comment related to this is to change our output to have two columns - one with the ontology id and one where we grab the full cell ontology label using `ontoProc`. 

> We could directly updated the scpca-project-celltype-metadata.tsv file itself, but with just one line per scpca_project_id. This seems to be the right move imo? Or at least better than below..

I agree that we can keep the overall nextflow part the same and just change the number of references per project. This can be done as a last step. I think the main focus here would be only to get `label.ont` and remove the other labels since we are less interested in that.

**Comment by sjspielman** [comment]
> I think this would simplify everything so that we only work with the `label.ont` moving forward, which is what we want.

👍 

Thanks for the feedback here! I'm going to split this issue up into 2 and assign myself to them - 
1.  update workflow to take only 1 label
  - involves update reference building itself.
  - probably still want (at least it can't hurt) to have an `opt` in the R script for this with default `label.ont`, given the use of a `label_type` param in the reference building workflow (and I suppose also use `label_type` param in the annotation workflow too!)
    -  I wonder if it would be helpful to have a separate celltyping config file where `label_type` and `celltype_refs_metafile` could live?
2. blocked by ^, updating the workflow to take only 1 model

**Comment by allyhawkins** [comment]
> I wonder if it would be helpful to have a separate celltyping config file where label_type and celltype_refs_metafile could live?

Right now we have the `celltype_refs_metafile` defined in the ccdl profile. I think we could leave it like that for now and add in `label_type` there? 
https://github.com/AlexsLemonade/scpca-nf/blob/4b4de3c8b10415288982aabfb0c31378270ec834/config/profile_ccdl.config#L16

**Comment by sjspielman** [comment]
This issue was broken out into #380 and #381, which were respectively closed by #382 and #383. All set!
